src/coreclr/src/pal/src/exception/machexception.cpp

```diff
@@ -356,11 +356,16 @@ PAL_ERROR CorUnix::CPalThread::DisableMachExceptions()
     return palError;
 }
 
+#if defined(HOST_AMD64) || defined(HOST_X86)
 // Since HijackFaultingThread pushed the context, exception record and info on the stack, we need to adjust the
 // signature of PAL_DispatchException such that the corresponding arguments are considered to be on the stack
 // per GCC64 calling convention rules. Hence, the first 6 dummy arguments (corresponding to RDI, RSI, RDX,RCX, R8, R9).
 extern "C"
 void PAL_DispatchException(DWORD64 dwRDI, DWORD64 dwRSI, DWORD64 dwRDX, DWORD64 dwRCX, DWORD64 dwR8, DWORD64 dwR9, PCONTEXT pContext, PEXCEPTION_RECORD pExRecord, MachExceptionInfo *pMachExceptionInfo)
+#elif defined(HOST_ARM64)
+extern "C"
+void PAL_DispatchException(PCONTEXT pContext, PEXCEPTION_RECORD pExRecord, MachExceptionInfo *pMachExceptionInfo)
```

Comment:
I think that we need to do a similar trick like on x64 and pass in all argument registers as dummy (so that we don't touch any of them).

Reply:
It wasn't clear why we had to leave the physical registers untouched as we are passing them in the context to the callee.  It looked like the amd64 stack trick wasn't needed.

Reply:
We will need to verify that stack unwinding in LLDB is correct when the PAL_DispatchException is on the stack once we can execute managed code.

Reply:
This seems to be working as is.